infra/blueprint-test/pkg/tft/terraform.go

```diff
@@ -481,22 +488,50 @@ func (b *TFBlueprintTest) DefaultApply(assert *assert.Assertions) {
 
 // Init runs the default or custom init function for the blueprint.
 func (b *TFBlueprintTest) Init(assert *assert.Assertions) {
+	// allow only single write as Terraform plugin cache isn't concurrent safe
+	if err := b.tftCacheMutex.Lock(); err != nil {
```

Comment:
Does lock wait to acquire the lock or throw an error if already locked?

Reply:
The doc's are minimal, but `Lock()` is supposed to block till the lock is acquired.  `TryLock()` would throw an error if already locked.

Reply:
Got it thanks!